Turn on findbugs and add annotation support

spacecowboy opened the pull request:
It was already there as a plugin. I updated the version and enabled it
to run during builds. Without failing them for now. As we fix the reported issues,
we can turn on "failOnError" for a per module basis, and finally for the entire project.

Also adds dependencies to the findbugs annotations library, so one can
use things like `@Nullable` and `@Nonnull`. These have been added to low level modules
and so should be available transitively in most other modules.

The first commit only changes pom-files (in I hope the minimal way possible)

The second commit does the necessary license fixes that entails. (updates all text files, this is auto-generated by building with `-Doverwrite`)

The third commit actually makes use of the annotation support. This means that the
following error (reported in #6335) becomes visible (will be fixed in a separate PR):

![selection_004](https://cloud.githubusercontent.com/assets/223655/12848975/3fee1fd2-cc1d-11e5-841a-74a794d03a9a.png)

Once we flip the switch, that error would actually fail the build (but not yet!)

spacecowboy committed:
Add @Nonnull annotations to Logging classes

lutovich commented:
![have-a-happy-day-015](https://cloud.githubusercontent.com/assets/1638145/12849382/ad248e18-cc1f-11e5-9628-01d498c6e818.gif)

spacecowboy committed:
Turn on findbugs and add annotation support

It was already there as a plugin. I updated the version and enabled it
to run during builds. Without failing them for now.

Also adds dependencies to the findbugs annotations library, so one can
use things like `@Nullable` and `@Nonnull`.

spacecowboy committed:
Update licenses to include FindBugs

MishaDemianenko commented:
@spacecowboy you need to update some license files

spacecowboy commented:
@MishaDemianenko working on it... they are all over the place!

spacecowboy committed:
Package licenses

spacecowboy committed:
Update manual licenses

spacecowboy committed:
Update manual LICENSE files
